python/cudf_polars/cudf_polars/experimental/benchmarks/pdsds_queries/q11.py

```diff
@@ -0,0 +1,288 @@
+                pl.col("c_preferred_cust_flag").alias("customer_preferred_cust_flag"),
+                pl.col("c_birth_country").alias("customer_birth_country"),
+                pl.col("c_login").alias("customer_login"),
+                pl.col("c_email_address").alias("customer_email_address"),
+                pl.col("d_year").alias("dyear"),
+                pl.col("year_total"),
+            ]
+        )
+    )
+
+
+def polars_impl(run_config: RunConfig) -> pl.LazyFrame:
+    """Query 11."""
+    # Load required tables
+    customer = get_data(run_config.dataset_path, "customer", run_config.suffix)
+    store_sales = get_data(run_config.dataset_path, "store_sales", run_config.suffix)
+    web_sales = get_data(run_config.dataset_path, "web_sales", run_config.suffix)
+    date_dim = get_data(run_config.dataset_path, "date_dim", run_config.suffix)
+
+    date_first = date_dim.filter(pl.col("d_year") == YEAR_FIRST)
+    date_second = date_dim.filter(pl.col("d_year") == YEAR_SECOND)
+
+    # Total store sales 2001
+    t_s_firstyear = create_year_total(
+        customer,
+        store_sales,
+        "ss_customer_sk",
+        "ss_sold_date_sk",
+        "ss_ext_list_price",
+        "ss_ext_discount_amt",
+        date_first,
+    ).select(
+        [
+            pl.col("customer_id").alias("s_first_customer_id"),
+            pl.col("year_total").alias("s_first_year_total"),
+        ]
+    )
+
+    # Total store sales 2002
+    t_s_secyear = create_year_total(
+        customer,
+        store_sales,
+        "ss_customer_sk",
+        "ss_sold_date_sk",
+        "ss_ext_list_price",
+        "ss_ext_discount_amt",
+        date_second,
+    ).select(
+        [
+            pl.col("customer_id").alias("s_sec_customer_id"),
+            pl.col("customer_first_name"),
+            pl.col("customer_last_name"),
+            pl.col("customer_birth_country"),
+            pl.col("year_total").alias("s_sec_year_total"),
+        ]
+    )
+
+    # Total web sales 2001
+    t_w_firstyear = create_year_total(
+        customer,
+        web_sales,
+        "ws_bill_customer_sk",
+        "ws_sold_date_sk",
+        "ws_ext_list_price",
+        "ws_ext_discount_amt",
+        date_first,
+    ).select(
+        [
+            pl.col("customer_id").alias("w_first_customer_id"),
+            pl.col("year_total").alias("w_first_year_total"),
+        ]
+    )
+
+    # Total web sales 2002
+    t_w_secyear = create_year_total(
+        customer,
+        web_sales,
+        "ws_bill_customer_sk",
+        "ws_sold_date_sk",
+        "ws_ext_list_price",
+        "ws_ext_discount_amt",
+        date_second,
+    ).select(
+        [
+            pl.col("customer_id").alias("w_sec_customer_id"),
+            pl.col("year_total").alias("w_sec_year_total"),
+        ]
+    )
+
+    # Join the tables and filter to get customers whose web and store spending grew from 2001 -> 2002
+    return (
```

Comment:
Note: I did think of computing the per-customer yearly totals, then doing the join to `customer` at the end to get the customer name/country. It would reduce the join tables, but it's not what DuckDB does. WDYT?

Reply:
From https://duckdb.org/2024/11/14/optimizers.html#summary-of-all-optimizers, it doesn't seem like DuckDB does it either.